src/PrestaShopBundle/Form/Admin/Sell/Catalog/AttributeType.php

```diff
@@ -0,0 +1,146 @@
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Form\Admin\Sell\Catalog;
+
+use PrestaShop\PrestaShop\Adapter\AttributeGroup\AttributeGroupDataProvider;
+use PrestaShop\PrestaShop\Adapter\AttributeGroup\Repository\AttributeGroupRepository;
+use PrestaShop\PrestaShop\Core\ConstraintValidator\Constraints\TypedRegex;
+use PrestaShop\PrestaShop\Core\ConstraintValidator\TypedRegexValidator;
+use PrestaShop\PrestaShop\Core\Domain\AttributeGroup\ValueObject\AttributeGroupId;
+use PrestaShop\PrestaShop\Core\Domain\AttributeGroup\ValueObject\AttributeGroupType;
+use PrestaShop\PrestaShop\Core\Domain\Language\ValueObject\LanguageId;
+use PrestaShop\PrestaShop\Core\Domain\Shop\ValueObject\ShopId;
+use PrestaShopBundle\Form\Admin\Type\ShopChoiceTreeType;
+use PrestaShopBundle\Form\Admin\Type\TranslatableType;
+use PrestaShopBundle\Form\Admin\Type\TranslatorAwareType;
+use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+use Symfony\Component\Form\Extension\Core\Type\ColorType;
+use Symfony\Component\Form\Extension\Core\Type\FileType;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Validator\Constraints\NotBlank;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+/**
+ * Form type for attribute add/edit
+ */
+class AttributeType extends TranslatorAwareType
+{
+    /**
+     * @var AttributeGroupRepository
+     */
+    private $attributeGroupRepository;
+
+    /**
+     * @var AttributeGroupDataProvider
+     */
+    private $attributeGroupDataProvider;
+
+    /**
+     * @var int
+     */
+    private $contextShopId;
+
+    /**
+     * @var int
+     */
+    private $contextLangId;
+
+    public function __construct(
+        TranslatorInterface $translator,
+        array $locales,
+        AttributeGroupRepository $attributeGroupRepository,
+        AttributeGroupDataProvider $attributeGroupDataProvider,
+        int $contextShopId,
+        int $contextLangId
+    ) {
+        parent::__construct($translator, $locales);
+
+        $this->attributeGroupRepository = $attributeGroupRepository;
+        $this->contextShopId = $contextShopId;
+        $this->attributeGroupDataProvider = $attributeGroupDataProvider;
+        $this->contextLangId = $contextLangId;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $attributeGroupId = $options['data']['attribute_group'];
```

Comment:
Instead of relying on the data itself which is quite dynamic you could add a required option on this form type. Then the form can absolutely never be instanciated without specifying this option, which makes sense as it is indeed required

We have a similar use case with product and combinations:
https://github.com/PrestaShop/PrestaShop/blob/030a383b64a5411275de02bb3b2708dfe5a8dc21/src/PrestaShopBundle/Form/Admin/Sell/Product/Combination/CombinationFormType.php#L110

For the combination in the controller the way to pass the require option is a bit complex (it uses a `FormOptionsProviderInterface`), but you don't need to go that far you can pass the option "manually" when the form is built, it's already handled except the value is passed via the `$data` parameter when it should be passed via the `$options` parameter:

```php
$attributeForm = $attributeFormBuilder->getFormFor($attributeId, [], ['attribute_group' => $attributeGroupId])
            ->handleRequest($request);
// instead of
$attributeForm = $attributeFormBuilder->getFormFor($attributeId, ['attribute_group' => $attributeGroupId])
            ->handleRequest($request);
```

Using explicit option bring a clearer and more stable contract/constraint on the form type